src/main/java/com/google/sps/servlets/SearchServlet.java

```diff
@@ -75,19 +75,47 @@ public JSONObject getMatchingCourses(HttpServletRequest request) {
   /* Create list of filters given parameters specified in request. */
   private List<Filter> getFilters(HttpServletRequest request, boolean fuzzy) {
     List<Filter> filters = new ArrayList<>();
-    String school = request.getParameter("school-name");
+    String school = request.getParameter("school-name").toLowerCase();
     Filter schoolFilter = new FilterPredicate("school", FilterOperator.EQUAL, school);
     if (!request.getParameter("courseName").isEmpty()) {
       String name = request.getParameter("courseName");
+      String[] splitName;
+      if (school.equals("mit")) { // Courses are formatted "department.courseNum"
+        splitName = name.split("\\.");
+      } else {
+        splitName = name.split(" "); // Courses are formatted "department courseNum"
+      }
       Filter nameFilter;
-      if (fuzzy) {
-        String department = name.split(" ")[0];
-        int courseNum = Integer.parseInt(name.split(" ")[1]);
+      if (fuzzy && splitName.length > 1) { // Check if course name is properly split
         List<String> courseNums = new ArrayList<>();
-        courseNums.add(department + " " + String.valueOf(courseNum + 1));
         courseNums.add(name);
-        courseNums.add(department + " " + String.valueOf(courseNum - 1));
-        nameFilter = new FilterPredicate("name", FilterOperator.IN, courseNums);
+        try {
+          String department = splitName[0];
+          int courseNum = Integer.parseInt(splitName[1]);
+          if (school.equals(
+              "mit")) { // Account for the fact that some course numbers have leading zeroes (e.g.
```

Comment:
For long comments like this (and often for comments in general), we tend to put them on their own lines to avoid weird wrapping like this.